posthog/migrations/0868_create_cohort_calculation_history.py

```diff
@@ -0,0 +1,53 @@
+# Generated by Django 4.2.22 on 2025-10-01 13:01
+
+import django.utils.timezone
+import django.db.models.deletion
+from django.db import migrations, models
+
+import posthog.models.utils
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("posthog", "0867_add_updated_at_to_feature_flags"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="CohortCalculationHistory",
+            fields=[
+                (
+                    "id",
+                    models.UUIDField(
+                        default=posthog.models.utils.uuid7, editable=False, primary_key=True, serialize=False
+                    ),
+                ),
+                ("filters", models.JSONField(help_text="Cohort filters/properties at time of calculation")),
+                ("count", models.PositiveIntegerField(blank=True, help_text="Number of people in cohort", null=True)),
+                (
+                    "started_at",
+                    models.DateTimeField(default=django.utils.timezone.now, help_text="When calculation started"),
+                ),
+                ("finished_at", models.DateTimeField(blank=True, help_text="When calculation finished", null=True)),
+                (
+                    "queries",
+                    models.JSONField(
```

Comment:
I think my only worry here is that queries might grow somewhat unbounded since it's a jsonb field and you could be storing many queries in production. Maybe worth monitoring somehow.